### PR TITLE
Algolia

### PR DESCRIPTION
**Pull request by Era-cell**
I tried fuse search and it was successful, please check if this works,
IF yes next will be:
searchbar positioning
Implement same for vocabulary page and dialect page
Customize fuseOptions much better

### CODE REVIEW

File: layouts/_default/index.json
Comment (jviotti) [inline layouts/_default/index.json:1]: What is this file? We shouldn't be committing generated files in this project. You can get Hugo to generate a JSON file automatically from the site content

File: config/_default/config.toml [hunk 1]
```diff
@@ -13,3 +13,4 @@ jsonschema = 'https://json-schema.org'
   [markup.highlight]
     lineNos = false
     noClasses = false
+
```
Comment (jviotti) [inline config/_default/config.toml:16]: Empty line?

File: assets/main.scss [hunk 2]
```diff
@@ -146,3 +146,16 @@ details:not([open]) summary i {
   @extend .d-inline-block;
   transform: rotate(-90deg); 
 }
+
+/* Customizing the search button color */
+.btn-outline-success {
```
Comment (jviotti) [inline assets/main.scss:151]: You should never override Bootstrap classes. Bootstrap provides a way to override main colors through SCSS variables and also supports plenty of helpers to customize colors, border colors, etc of arbitrary elements.

File: assets/main.js [hunk 3]
```diff
@@ -5,3 +5,68 @@ import Tooltip from './vendor/bootstrap/js/src/tooltip.js';
 for (const element of document.querySelectorAll('[data-bs-toggle="tooltip"]')) {
   new Tooltip(element);
 }
+
+// Fuse search options
+var fuseOptions = {
```
Comment (jviotti) [inline assets/main.js:10]: `var` is an old and no longer used JavaScript construct. You should use `const` or `let`

File: layouts/partials/navigation.html [hunk 4]
```diff
@@ -5,11 +5,18 @@
         height="42" width="42" class="me-1">
       <span class="align-middle">JSON Schema Docs</span>
     </a>
-    <div class="navbar-nav">
+
+    <form class="d-flex justify-self-end" id="search-form">
+      <input class="form-control me-2" type="search" name="q" id="search-query" placeholder="Search" aria-label="Search">
+      <button class="btn btn-outline-success" type="submit">Search</button>
```
Comment (jviotti) [inline layouts/partials/navigation.html:11]: Probably no need to have a submit button, as it is not a form you would be submitting anyway. Ideally we just have the search input and we search as the user types?

File: layouts/partials/navigation.html [hunk 5]
```diff
@@ -5,11 +5,18 @@
...
+    </form>
+
+    <div class="navbar-nav d-flex ">
```
Comment (jviotti) [inline layouts/partials/navigation.html:14]:
```suggestion
    <div class="navbar-nav d-flex">
```

**Review by jviotti — CHANGES_REQUESTED** [review]
I left a few comments have various comments about the JavaScript part, but it doesn't seem to work for me at all. I don't see anything if I type on the search bar or submit, as if nothing happens at all. Does it work for you?

File: layouts/_default/index.json
Comment (Era-cell) [reply layouts/_default/index.json:1]: Yeah, I saw that auto generation didn't work for me so I copied the information manually. Let me try it again to auto-generate

File: config/_default/config.toml (hunk 1 above)
Comment (Era-cell) [reply config/_default/config.toml:16]: So, .toml files shouldn't have empty lines in the end?? I and any config file?.. I'll change it,

File: layouts/partials/navigation.html (hunk 4 above)
Comment (Era-cell) [reply layouts/partials/navigation.html:11]: " Search as the user inputs" means we will keep redirecting as user inputs or keep changing suggestions like in JSON Schema website as user inputs ?? 1 or 2?

**Comment by Era-cell** [comment]
> I left a few comments have various comments about the JavaScript part, but it doesn't seem to work for me at all. I don't see anything if I type on the search bar or submit, as if nothing happens at all. Does it work for you?

This is the recording of the working : it redirects directly, ..

https://github.com/Intelligence-AI/learnjsonschema.com/assets/77973415/0a2770e9-b3bc-4b21-be66-d2db978f1e4d

**Comment by jviotti** [comment]
I don't get the dropdown at all. What browser are you using? Maybe try other ones?

https://github.com/Intelligence-AI/learnjsonschema.com/assets/2192773/078c8a95-3f84-4b87-af38-12c08ab38f5c

File: layouts/partials/navigation.html (hunk 4 above)
Comment (jviotti) [reply layouts/partials/navigation.html:11]: No redirection. Exactly like in the JSON Schema official website: we dynamically change the dropdown with the search results

File: config/_default/config.toml (hunk 1 above)
Comment (jviotti) [reply config/_default/config.toml:16]: I don't think any file format as a requirement of an empty line at the end that I'm aware of

**Comment by Era-cell** [comment]
> I don't get the dropdown at all. What browser are you using? Maybe try other ones?
> 
>  Untitled.mov

That dropdown isn't code defined, yeah its because of the browser - Chrome browser. 

> I don't get the dropdown at all. What browser are you using? Maybe try other ones?
> 
>  Untitled.mov

That dropdown is a browser behavior. Right now it works only if the search button is clicked. Let's implement listing functionality too..

File: layouts/partials/navigation.html (hunk 4 above)
Comment (jviotti) [reply layouts/partials/navigation.html:11]: You should be able to doing with plain JavaScript + Bootstrap. i.e. load a Bootstrap dropdown upon search results. Or if you can create a JavaScript function that gives you the search results given a search term as an array I can help doing the UI

**Comment by Era-cell** [comment]
@jviotti I have made the major changes, styling and html should be polished.
Currently(functionality):
1. I have made it such that it clears everything on clicking outside, maybe we can give top suggestions
2. listing out top 10 results.
Change suggestions on these?

**Comment by jviotti** [comment]
Seems to be shaping up but still doesn't work for me. It cannot find the JSON index file

Screenshot 2024-05-28 at 9 51 40 AM

Also, can you resolve the git conflicts?

**Comment by jviotti** [comment]
Woah! Super cool already! One small UI suggestion: when displaying a keyword, can you also showcase what dialect and vocabulary they belong to?

**Comment by jviotti** [comment]
Can you extract the parts of this PR that generate the JSON index as a separate PR? I think we can get that merged now while we focus on polishing the rest

File: layouts/_default/baseof.html [hunk 6]
```diff
@@ -34,11 +34,20 @@
...
+    {{ end }}
+    
     {{ partial "footer.html" . }}
-
+    
     {{ $js := resources.Get "main.js" | js.Build (dict "minify" true "target" "es2015") }}
-    <script defer src="{{ $js.Permalink }}"></script>
+    <script async src="{{ $js.Permalink }}"></script>
+    
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.6.2/fuse.min.js" integrity="sha512-Nqw1tH3mpavka9cQCc5zWWEZNfIPdOYyQFjlV1NvflEtQ0/XI6ZQ+H/D3YgJdqSUJlMLAPRj/oXlaHCFbFCjoQ==" crossorigin="anonymous" referrerpolicy="no-referrer"></script>
```
Comment (jviotti) [inline layouts/_default/baseof.html:48]: Can we pull these dependencies using NPM on `package.json`?

**Comment by Era-cell** [comment]
> Woah! Super cool already! One small UI suggestion: when displaying a keyword, can you also showcase what dialect and vocabulary they belong to?

Okay.. I got the reason for it . For the UI shall I include both of them in right as a smaller size text ? I may have to override bootstrap I guess

**Comment by jviotti** [comment]
Not sure about the style. Give it your best shot while changing the least amount of possible in Bootstrap and then I'll help!

**Comment by Era-cell** [comment]
@jviotti I have made the changes, I have updated package.json too, should I have another PR for index.json... 100%?

**Comment by jviotti** [comment]
Yeah, let's do a separate PR. A sequence of smaller PRs is always better than 1 bigger PR! At least we have the JSON part out of the way, and we focus on the rest.

**Pull request closed without merging**